src/js/components/LayerPanel/TerraIControls.js

```diff
@@ -96,16 +100,19 @@ export default class TerraIControls extends Component {
 
   didSetEndDate = ({select}) => {
     if (select) {
-      layerActions.updateTerraIEndDate(new Date(select));
-      this.updateDateRange();
+      const endDate = new Date(select);
+      layerActions.updateTerraIEndDate(endDate);
+      this.updateDateRange(this.props.startDate, endDate);
       if (this.fromPicker && this.toPicker) {
         this.fromPicker.set('max', this.toPicker.get('select'));
       }
     }
   };
 
-  updateDateRange = () => {
-    const {layer, startDate, endDate} = this.props;
+  updateDateRange = (startDate, endDate) => {
+    const {layer} = this.props;
+    startDate = startDate.constructor === Date ? startDate : this.state.min;
+    endDate = endDate.constructor === Date ? endDate : this.state.max;
```

Comment:
same thing happening here. Since this is called once for each time the start or end is set separately, we will always have one date and one empty object when the clear button is pressed. To fix that we just do a check and return either the param passed or the min/max in state.